TrustModel.md

```diff
@@ -0,0 +1,426 @@
+
+In considering personas we recognise that a trust boundary exists between each persona and we
+explore how the least privilege principle (as described in Cloud Native security Paper, Page 36 
+[^2] ) should apply to any actions which cross these boundaries.
+
+ Confidential containers can provide enhancements to ensure that the 
+expected  code/containers are the only code that can operate over the data. However any  
+vulnerabilities within this code are not mitigated by using confidential containers, the Cloud  
+Native Security Whitepaper details Lifecycle aspects that relate to the security of the code 
+being placed into containers such as Static/Dynamic Analysis, Security Tests, Code Review etc 
+which must still be followed. 
+
+![Personas Model](./images/PersonaModel.svg)
+
+Anyone of these personas could attempt to perform malicious actions.
+
+### Cloud Operator
+This persona has privileges within the Cloud/Environment Infrastructure.
+They are responsible for availability of infrastructure used by cloud native environment
+- Has access to the physical hardware.
+- Has access to the process involved in deploying VSI's.
+- Has control over firmware versions
+- Has control over TEE hardware availability/type
+
+Example : Cloud Service Providers IaaS SRE's
+
+### Orchestration Operator
+This persona has privileges within the Orchestration/Cluster.
+They are responsible for deploying a solution into a particular cloud native environment and 
+managing the orchestration environment.
+For Managed k8s cluster this would also include the administration of the cluster control plane.
+- controls availability of service
+- controls webhooks and deployment 
+- controls availability of cluster resources (data/networking/storage)
+- deploys protected boot image
+
+Example : A kubernetes administrator responsible for deploying pods to a cluster and 
+maintaining the cluster.
+
+### Workload Provider
+This persona has control over the code, containers and pods used to provide the Application or 
+Workload.
+They define and build an application or solution to provide business value to customers and 
+clients.
+The workload provider and end user could be part of same company/organisation but 
+following the least privilege principle the application provider should not be able to view or 
+manipulate end user data without informed consent.
+- Owner of business logic containers. These may contain proprietary algorithms, models or secrets
+- Needs to prove to customer aspects of compliance
+- Defines what the solution requires in order to run and maintain compliance (resources, utility 
+  containers/services,  storage)
+- provide the protected bootimage, or designate a trusted party to provide the protected 
+  bootimage; 
+- provide the attestation verification service, or designate a trusted 
+  party  to provide the attestation verification service.
+
+Example : 3rd party software vendor
```

Comment:
> Currently whoever provides a confidential container must have access to the keys that encrypted/decrypt it. I'm not sure how a 3rd party vendor could do this on behalf of a client.

The initial image encryption can determine authorizations to decrypt the image. Theoretically, the authorized recipients may even not the one who launches the initial image encryption. In addition, any one of authorized recipients can add recipients to an already encrypted image.

So it is possible that a 3rd party software vendor may play this role to launch the initial image encryption.

> More fundamentally, we currently have no practical separation between the data provider and the workload provider. We have been referring to both as the guest-owner or simply the client/tenant. I understand how it might be useful to conceive of these as two different parties, but it feels out of sync with what we have built and discussed. There is a monolithic trust domain inside the enclave. I'm not aware of any proposals to change that and I am not sure how we would.

Actually confidential computing doesn't provide the isolation between the data provider and the workload provider in such granularity (or not necessary). A programmable hardware TEE must at least provide data confidentiality, data integrity and code integrity [1, Chapter 3 Trusted Execution Environments (TEEs)], which implies the workload provider can access and manipulate the data provided by data provider. The trust between them is established through other ways [1, the last paragraph at page 10]. 

[1] https://confidentialcomputing.io/wp-content/uploads/sites/85/2021/03/CCC-Tech-Analysis-Confidential-Computing-V1.pdf

Reply:
> The initial image encryption can determine authorizations to decrypt the image. Theoretically, the authorized recipients may even not the one who launches the initial image encryption. In addition, any one of authorized recipients can add recipients to an already encrypted image.

A 3rd party could provide an encrypted container as long as they also provide the key for it. The client/tenant must then trust that 3rd party. Fundamentally, whoever encrypts the container must be trusted. So while a client could use a trusted third party to get encrypted containers, I don't think a 3rd party software vendor is the best example of a workload provider.

> Actually confidential computing doesn't provide the isolation between the data provider and the workload provider in such granularity (or not necessary).

Yes, this is exactly my point. Should the trust model contain separate roles for workload provider and data provider if the underlying hardware/software doesn't make a distinction between them and they must be mutually trusting?

Reply:
> A 3rd party could provide an encrypted container as long as they also provide the key for it. The client/tenant must then trust that 3rd party. Fundamentally, whoever encrypts the container must be trusted. So while a client could use a trusted third party to get encrypted containers, I don't think a 3rd party software vendor is the best example of a workload provider.

I think 3rd party could not provide the key, just using the authorized client/tenant's pubkey to encrypt the image. But I fully agree on the remaining stuffs. The client/tenant must trust that 3rd party. As an example to describe the role of workload provider, I admit 3rd party is complex a bit because it is related to transitive trust.

> Yes, this is exactly my point. Should the trust model contain separate roles for workload provider and data provider if the underlying hardware/software doesn't make a distinction between them and they must be mutually trusting?

This is another case relating to transitive trust. Obviously the data owner needs to delegate the trust to workload provider. I think two roles are easy to describe some high level CC's application scenarios such as FaaS.

Reply:
In my experience the distinction between workload provider and data owner exists and is in general use, especially as we consider kubernetes sidecars and software as a service. As such while I agree we are not advocating that the TEE can distinguish between the data owner or the workload provider acknowledging these personas exist is important. We promise confidential computing and protection of data if we bundle these personas together when others think of them as different then we are not being clear with people when our trust model requires some other means of establishing trust between them.

Beyond this I have added a conclusion section to the new PR (https://github.com/confidential-containers/documentation/pull/24) specifically targeting Trust Model Personas and included an example to highlight that the trust between workload provider and data owner is not always complete in both directions and nor does it have to be.

I think our previous discussion on container signatures v encryption is particularly relevant here. 
As a data owner I can use signatures to establish trust in a particular workload and protect my data with an encryption secret. So I am not trusting the workload provider just one workload and I use confidential containers to ensure integrity of my trust (i.e signatures) . The data owner controls the TEE and deployment including storage used and can revoke encryption of data as they control the data storage.
As a workload provider I am using container encryption to protect my containers, I will allow a data owner to use my workload but not see the underlying code. Here the workload provider controls the TEE and deployment and potentialy the storage used.


Transitive trust is not something we have gone into in detail, I think it makes more sense in the context of specific threats to explore that though. So something for later.